### PR TITLE
Refactor DropRoleTask

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
Re-attempt https://github.com/crate/crate/pull/15742

## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by jeeminso** [comment]
`DropRoleRequest.ifExists` is actually unused and can be removed but could also keep it for future refactoring.

File: server/src/main/java/io/crate/role/DropRoleTask.java [hunk 1]
```diff
@@ -60,12 +60,11 @@ public ClusterState execute(ClusterState currentState) throws Exception {
...
     @Override
-    protected WriteRoleResponse newResponse(boolean acknowledged) {
-        return new WriteRoleResponse(acknowledged, alreadyExists);
+    protected AcknowledgedResponse newResponse(boolean acknowledged) {
+        return new AcknowledgedResponse(acknowledged && alreadyExists);
     }
 
-    @VisibleForTesting
-    static boolean dropRole(Metadata.Builder mdBuilder, String roleNameToDrop) {
+    private static boolean dropRole(Metadata.Builder mdBuilder, String roleNameToDrop) {
```
Comment (matriv) [inline server/src/main/java/io/crate/role/DropRoleTask.java:67]: Why did you change that? The unit tests now are a bit more cumbersome. Do we gain something to test the `execute` instead of the `dropRole` method?

**Review by matriv — APPROVED** [review]
thank you!

File: server/src/main/java/io/crate/role/DropRoleTask.java (hunk 1 above)
Comment (matriv) [reply server/src/main/java/io/crate/role/DropRoleTask.java:67]: yep, it's fine, it doesn't add too much boilerplate to the tests, thx.
